clients/src/main/java/org/apache/kafka/common/metrics/stats/SampledStat.java

```diff
@@ -50,10 +50,13 @@ public void record(MetricConfig config, double value, long timeMs) {
             sample = advance(config, timeMs);
         update(sample, config, value, timeMs);
         sample.eventCount += 1;
+        sample.lastEventMs = timeMs;
     }
 
     private Sample advance(MetricConfig config, long timeMs) {
-        this.current = (this.current + 1) % config.samples();
+        // need to keep one extra sample (see purgeObsoleteSamples() logic)
```

Comment:
I slightly changed the suggestion, please check if it's readable. Because technically this extra space is not to remember the last recording time, it's to keep the entire extra sample with all collected data.

Reply:
your comment is great! thanks!